src/types.rs

```diff
@@ -415,6 +422,16 @@ impl Value {
         }
     }
 
+    fn resolve_date(self) -> Result<Self, Error> {
+        match self {
+            // TODO: does this need to also handle Value::Int/Long for schema evolution or some other reason?
```

Comment:
Mercifully the spec doesn't seem to say anything about schema evolution with dates.